modules/xfd.ts

```diff
@@ -0,0 +1,2218 @@
+			this.params.intialContrib = null;
+			return $.Deferred().resolve();
+		}
+		return this.notifyTalkPage(this.params.initialContrib);
+	}
+
+	// Should be called after notifyTalkPage() which may unset this.params.intialContrib
+	addToLog() {
+		let params = this.params;
+
+		if (!Twinkle.getPref('logXfdNominations') ||
+			Twinkle.getPref('noLogOnXfdNomination').indexOf(params.venue) !== -1) {
+			return $.Deferred().resolve();
+		}
+
+		var usl = new Morebits.userspaceLogger(Twinkle.getPref('xfdLogPageName'));// , 'Adding entry to userspace log');
+
+		usl.initialText =
+			"This is a log of all [[WP:XFD|deletion discussion]] nominations made by this user using [[WP:TW|Twinkle]]'s XfD module.\n\n" +
+			'If you no longer wish to keep this log, you can turn it off using the [[Wikipedia:Twinkle/Preferences|preferences panel]], and ' +
+			'nominate this page for speedy deletion under [[WP:CSD#U1|CSD U1]].' +
+			(Morebits.userIsSysop ? '\n\nThis log does not track XfD-related deletions made using Twinkle.' : '');
+
+		usl.changeTags = Twinkle.changeTags;
+		return usl.log(this.getUserspaceLoggingText(), this.getUserspaceLoggingEditSummary());
+	}
+
+	getUserspaceLoggingEditSummary() {
+		return 'Logging ' + utils.toTLACase(this.params.venue) + ' nomination of [[:' + Morebits.pageNameNorm + ']].';
+	}
+
+	getUserspaceLoggingText(): string {
+		let params = this.params;
+
+		// If a logged file is deleted but exists on commons, the wikilink will be blue, so provide a link to the log
+		var fileLogLink = mw.config.get('wgNamespaceNumber') === 6 ? ' ([{{fullurl:Special:Log|page=' + mw.util.wikiUrlencode(mw.config.get('wgPageName')) + '}} log])' : '';
+		// CFD/S and RM don't have canonical links
+		var nominatedLink = params.discussionpage ? '[[' + params.discussionpage + '|nominated]]' : 'nominated';
+
+		var appendText = '# [[:' + Morebits.pageNameNorm + ']]:' + fileLogLink + ' ' + nominatedLink + ' at [[WP:' + params.venue.toUpperCase() + '|' + utils.toTLACase(params.venue) + ']]';
+
+		appendText += this.getUserspaceLoggingExtraInfo();
+
+		if (params.initialContrib && params.notifycreator) {
+			appendText += '; notified {{user|1=' + params.initialContrib + '}}';
+		}
+		appendText += ' ~~~~~';
+		if (params.reason) {
+			appendText += "\n#* '''Reason''': " + Morebits.string.formatReasonForLog(params.reason);
+		}
+		return appendText;
+	}
+
+	getUserspaceLoggingExtraInfo() {
+		return '';
+	}
+
+}
+
+class Afd extends XfdMode {
+	static venueCode = 'afd';
+	static venueLabel = 'AfD (Articles for deletion)';
+
+	static isDefaultChoice() {
+		return mw.config.get('wgNamespaceNumber') === 0;
+	}
+
+	getFieldsetLabel() {
+		return 'Articles for deletion';
+	}
+
+	getVenueWarning() {
+		if (mw.config.get('wgNamespaceNumber') !== 0) {
+			return 'AfD is generally appropriate only for articles.';
+		} else if (mw.config.get('wgIsRedirect')) {
+			return 'Please use RfD for redirects.';
+		}
+	}
+
+	getMenuTooltip() {
```

Comment:
same question about static value for menu tooltip

Reply:
Same as above. It's invoked on an object reference rather than a class reference.